### PR TITLE
Queue beanstalk patch 1

### PR DESCRIPTION
**Pull request by korolev-d-l**
php unit-tests/manual-unit.php BeanstalkTest.php BeanstalkTest
Testing BeanstalkTest [OK]

Need rebuild phalcon for apply change.

### CODE REVIEW

**Comment by sergeyklay** [comment]
-1 There are many constants to add

**Comment by Chameleon-m** [comment]
@sergeyklay Zephir, насколько я знаю, при компиляции подставляет значение констант, там где они используются. 
В  чем недостаток их количества?

**Comment by sergeyklay** [comment]
memory for constants is allocated always no matter if you use the class or not

**Comment by Chameleon-m** [comment]
it is a very small memory allocation, and only 1 time.
Ок.

**Comment by Chameleon-m** [comment]
@sergeyklay

**Comment by sergeyklay** [comment]
As you can see tests fails

**Comment by Chameleon-m** [comment]
@sergeyklay 
To start beanstalkd, I need to add these lines in .travis.yml?
before_install:
 - ...
 - sudo apt-get install beanstalkd
 - beanstalkd -l 127.0.0.1 -p 11300

**Comment by sergeyklay** [comment]
**.travis.yml**
```yaml
sudo: required
dist: precise
group: edge

env:
  global:
    - TEST_BT_HOST="127.0.0.1"
    - TEST_BT_PORT="11300"


before_install:
 - bash tests/_ci/install_beanstalkd.sh
```

**tests/_ci/install_beanstalkd.sh**: https://github.com/sergeyklay/cphalcon/blob/updating_tests/tests/_ci/install_beanstalkd.sh

**Comment by Chameleon-m** [comment]
@sergeyklay thanks, left to deal with php 5.4

**Comment by Chameleon-m** [comment]
@sergeyklay Tell me please, why so?

I looked log, in 5.4 
configuration option "php_ini" is not set to php.ini location
In 5.5 and 5.6 - good
Extension yaml enabled in php.ini

**Comment by sergeyklay** [comment]
Try to install yaml via https://github.com/sergeyklay/cphalcon/blob/updating_tests/tests/_ci/install_prereqs.sh

**Comment by Green-Cat** [comment]
Do we really need this many constants?

**Comment by Chameleon-m** [comment]
@Green-Cat yes, for use with php

**Comment by Chameleon-m** [comment]
@sergeyklay thanks, I'm sorry for a lot of commits . ready!

File: unit-tests/ci/install_prereqs.sh [hunk 1]
```diff
@@ -19,6 +19,10 @@ CFLAGS="-O1 -g3 -fno-strict-aliasing" pecl install imagick < /dev/null &
 CFLAGS="-O1 -g3 -fno-strict-aliasing" pecl install yaml < /dev/null &
 # CFLAGS="-O2 -g3 -fno-strict-aliasing" pecl upgrade mongo < /dev/null &
 
+if [ "$(php -r 'echo substr(PHP_VERSION, 0, 3);')" = "5.4" ]; then
```
Comment (Green-Cat) [inline unit-tests/ci/install_prereqs.sh]: This is not needed, support for php 5.3 has been dropped in the 2.1.x branch.

**Comment by andresgutierrez** [comment]
I think we should not add that many constants, all constants in extensions are allocated always per request.

**Comment by Green-Cat** [comment]
@andresgutierrez Me too. I also don't really see the benefit from having them in the first place. The constants essentially contain a string version of themselves, so not much better developer experience than just using string commands IMO.

**Comment by sergeyklay** [comment]
At least all the `CMD_` and `MSG_` constants need to be removed

**Comment by Chameleon-m** [comment]
good, but it's at the expense of clarity and usability...

**Comment by sergeyklay** [comment]
Also rebase onto 2.1.x please

**Comment by sergeyklay** [comment]
Note, we use tabs for `*.zep` files, not spaces.

**Comment by sergeyklay** [comment]
@Chameleon-m Could you please rebase?

**Comment by Chameleon-m** [comment]
@sergeyklay I will try this week ... unfortunately little free time

**Comment by Chameleon-m** [comment]
see https://github.com/phalcon/cphalcon/pull/11552

**Pull request closed without merging**